cw/util/device.py

```diff
@@ -152,6 +152,10 @@ def initialize_scope(self, scope_gain, num_samples, offset, pll_frequency):
         scope.io.tio2 = "serial_rx"
         scope.io.hs2 = "disabled"
 
+        # Make sure that clkgen_locked is true.
+        if husky:
+            scope.clock.clkgen_src = 'extclk'
+
```

Comment:
Is this right? The comment seems not to be in line with the code. Also, we already set `scope.clock.clkgen_src = 'extclk'` above on Line 127.

Reply:
Without this code line, the PLL was not locked making the ADC unable to lock itself. According to [this](https://forum.newae.com/t/adc-freq-setting/3108/2), again setting `clkgen_src` fixes the issue, which also worked on my site.

Reply:
agree, odd, seems to be done twice now. does the if in L125 work properly?
I'd guess it evaluates false there even on husky?

Reply:
I think this is probably what prevented me from separating CW310 and Husky into two separate classes ... so I'd be interested :)

Reply:
thx, so L130 is in fact executed I understand.
tanks again - maybe will try split the two CW310 and Husky into two separate classes instead of OpenTitan in device.py later with this fix